app/src/main/java/free/rm/skytube/gui/fragments/PreferencesFragment.java

```diff
@@ -104,10 +113,46 @@ public void onPause() {
 
 	@Override
 	public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
-		// If the user changed the Default Tab Preference, update the summary to show the new default tab
-		if(key != null && key.equals(getString(R.string.pref_key_default_tab))) {
-			ListPreference defaultTabPref = (ListPreference)findPreference(key);
-			defaultTabPref.setSummary(String.format(getString(R.string.pref_summary_default_tab), defaultTabPref.getEntry()));
+		if(key != null) {
+			if (key.equals(getString(R.string.pref_key_default_tab))) {
+				// If the user changed the Default Tab Preference, update the summary to show the new default tab
+				ListPreference defaultTabPref = (ListPreference) findPreference(key);
+				defaultTabPref.setSummary(String.format(getString(R.string.pref_summary_default_tab), defaultTabPref.getEntry()));
+			} else if (key.equals(getString(R.string.pref_youtube_api_key))) {
+				// Validate the entered API Key when saved (and not empty), with a simple call to get the most popular video
+				final EditTextPreference youtubeAPIKeyPref = (EditTextPreference) findPreference(getString(R.string.pref_youtube_api_key));
+				if(!youtubeAPIKeyPref.getText().isEmpty()) {
```

Comment:
I'm getting a NullPointerException on Android 7.0 when I enter an invalid key such as "123"

```java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.isEmpty()' on a null object reference```

This can be easily resolved if we also check whether `youtubeAPIKeyPref.getText()` is null.